src/frontend/src/lib/components/dapps/DappsExplorerSignedIn.svelte

```diff
@@ -0,0 +1,80 @@
+<script lang="ts">
+	import { nonNullish } from '@dfinity/utils';
+	import DappCard from '$lib/components/dapps/DappCard.svelte';
+	import DappModal from '$lib/components/dapps/DappModal.svelte';
+	import DappPromoBanner from '$lib/components/dapps/DappPromoBanner.svelte';
+	import Button from '$lib/components/ui/Button.svelte';
+	import { modalDAppDetails } from '$lib/derived/modal.derived';
+	import { i18n } from '$lib/stores/i18n.store';
+	import { modalStore } from '$lib/stores/modal.store';
+	import {
+		type DappDescription,
+		dAppDescriptions,
+		type FeaturedDappDescription
+	} from '$lib/types/dappDescription';
+	import { replacePlaceholders } from '$lib/utils/i18n.utils';
+
+	let selectedTag: string | null = null;
+	const featuredDapp: FeaturedDappDescription | undefined = dAppDescriptions.find(
```

Comment:
- Shouldn't it be a `filter` - i.e. aren't multiple `featuredDapps` possible?

- the cast is needed? `dAppDescriptions` is not strongly typed enough?

- nitpick: not against fancy deconstruction

Reply:
- For the moment there will be only one featured dapp according to Artem.
- seems as it lints - my IDEA is not happy, but that's ok.
- I don't understand this, do you want deconstruction inside the find lambda?

Reply:
> For the moment there will be only one featured dapp according to Artem.

Can you add a comment or create a function - like "selectFirstFeaturedDapp" - to make this requirement explicit?

> deconstruction

```
// For the moment only the first featured dapp is highlighted
const selectFirstFeaturedDapp = (): FeaturedDappDescription | undefined => 
  dAppDescriptions.find(({featured, screenshots}) => featured && nonNullish(screenshots) && screenshots.length);

const featuredDapp = selectFirstFeaturedDapp()
```